stubs/Pillow/PIL/ImageTk.pyi

```diff
@@ -1,18 +1,52 @@
-from _typeshed import Incomplete
+import _tkinter
+import tkinter
+from _typeshed import ReadableBuffer, StrOrBytesPath, SupportsRead
 from typing import Any
 
-class PhotoImage:
-    tk: Any
-    def __init__(self, image: Incomplete | None = ..., size: Incomplete | None = ..., **kw) -> None: ...
+from PIL.Image import Image, _Box, _Mode, _Size
+
+class PhotoImage(tkinter._PhotoImageLike):
```

Comment:
This won't work very well with how third-party stubs are released. When running a type checker, you typically have a new version of third-party stubs (released every 3 hours if pull requests have been merged) and an old version of stdlib (comes with the type checker, mypy is released less than once in a month I think). So any users who upgrade their `pillow-stubs` package wouldn't actually get the protocol in  `tkinter/__init__.pyi`.

My experience is that mypy tends to hide errors coming from stub packages you have installed, and treat anything unknown as `Any`, so it would think that `ImageTk.PhotoImage` inherits from `Any` and therefore report no errors for trying to do `some_pil_photoimage.attribute_that_doesnt_exist`.

But this is really a temporary problem that will go away as new versions of type checkers are released.

Reply:
That is very true, I didn't think about the different release schedules. Not sure what to do there.

Reply:
Me neither. Both options have their own advantages and disadvantages :)

I'll probably merge this in a few days, unless someone comes up with a better idea or some reasoning to prefer the `width()`,`height()` protocol over fake classes.